tests/test-buttercup.el

```diff
@@ -881,7 +894,52 @@
       (it "throws an error when called"
         (expect (test-function 1 2)
                 :to-throw
-                'error "Stubbed error")))))
+                'error "Stubbed error")))
+
+    (describe "error-recording functionality"
+      (before-each
+        (spy-on 'test-function-throws-on-negative :and-call-through))
+
+      (it "records the function as called even if it throws an error"
+        (expect (test-function-throws-on-negative -5) :to-throw)
+        (expect (buttercup--apply-matcher
+                 :to-have-been-called
+                 (list (lambda () 'test-function-throws-on-negative)))
+                :to-be
+                t))
+
+      (it "counts both successful calls and calls that threw errors"
+        (test-function-throws-on-negative 5)
+        (expect (test-function-throws-on-negative -5) :to-throw)
+        (expect (buttercup--apply-matcher
+                 :to-have-been-called-times
+                 (make-list-of-closures '(test-function-throws-on-negative 2)))
+                :to-equal t))
+
+      (it "records args to the function whether it throw an error or not"
+        (test-function-throws-on-negative 5)
+        (expect (test-function-throws-on-negative -5) :to-throw)
+        (expect (buttercup--apply-matcher
+                 :to-have-been-called-with
+                 (make-list-of-closures '(test-function-throws-on-negative 5)))
+                :to-be
+                t)
+        (expect (buttercup--apply-matcher
+                 :to-have-been-called-with
+                 (make-list-of-closures '(test-function-throws-on-negative -5)))
+                :to-be
+                t))
+
+      (it "records the signal thrown by a call to the function"
+        (test-function-throws-on-negative 5)
+        (expect (test-function-throws-on-negative -5) :to-throw 'error)
+        (expect (spy-context-thrown-signal
+                 (spy-calls-first 'test-function-throws-on-negative))
+                :to-be nil)
+        (expect (car
```

Comment:
Why not check the entire signal like you do in the example in the next commit.

Reply:
Fixed in https://github.com/jorgenschaefer/emacs-buttercup/pull/146/commits/046031076bd7a53ab4d5a6dd873941a18c7d612f.